api/subjects/urls.py

```diff
@@ -0,0 +1,11 @@
+from django.conf.urls import url
+
+from api.subjects import views
+
+app_name = 'osf'
+
+urlpatterns = [
+    url(r'^$', views.SubjectList.as_view(), name=views.SubjectList.view_name),
```

Comment:
I don't think we need the root subjects list -- pretty sure we'll only ever list/filter subjects by provider. The corresponding `/v2/taxonomies/` endpoint was already deprecated, I assume for the same reason.